wp-content/themes/core/blocks/core/posttemplate/style.pcss

```diff
@@ -0,0 +1,50 @@
+/**
+ * Styles specific to this block
+ */
+
+/**
+ * Vertical (list layout) post template spacing
+ */
+.wp-block-post-template:not(.is-layout-grid) .wp-block-post + .wp-block-post {
+	margin-top: var(--spacer-40);
+}
+
+/**
+ * Horizontal (grid layout) post template spacing / media queries
+ *
+ * Essentially the calculations take the amount of columns and subtract
+ * the amount of spacers divided by the amount of columns. This gets the column
+ * width. The calculations subtract an extra pixel due to calc not enjoying our
+ * use of clamped spacer variables.
+ */
+.wp-block-post-template.is-layout-grid {
+	gap: var(--spacer-40);
+
+	&.columns-2 {
+		grid-template-columns: 1fr;
+
+		@media (--mq-wp-medium) {
+			grid-template-columns: repeat(2, minmax(0, 1fr));
+		}
+	}
+
+	&.columns-3 {
+		grid-template-columns: 1fr;
+
+		@media (--viewport-full) {
+			grid-template-columns: repeat(3, minmax(0, 1fr));
+		}
+	}
+
+	&.columns-4 {
+		grid-template-columns: 1fr;
+
+		@media (--mq-wp-medium) {
+			grid-template-columns: repeat(2, minmax(0, 1fr));
+		}
+
+		@media (--viewport-full) {
+			grid-template-columns: repeat(4, minmax(0, 1fr));
+		}
+	}
```

Comment:
A couple things:
* `grid-template-columns: 1fr;` could be defined within the top-most selector since it's applicable to all the column-count conditions.
* Rather than repeating the actual style definition for `grid-template-columns`, I'd suggest setting a css custom prop for the number of columns that is updated within the class and/or viewport size/media query blocks? It could  default to 2, reducing this to only the conditionals for 3 columns at full and 4 columns at full. Something like `grid-template-columns: repeat(var(--post-template-column-count), minmax(0, 1fr));`

Reply:
I've updated this. I had to add an extra breakpoint to the 3 columns layout though, because it needs to be one column at medium width instead of 2.